### DISCUSSION TITLE
Labwc equivalent for hyperctl/ riverctl: Accessing workspace and window information via IPC?

### DISCUSSION THREAD

**Discussion by AnirvanB** [labwc/labwc#2199] [General]
I've been using hyperctl and riverctl for tiling window managers and was wondering if there’s a similar tool for labwc, even though it's a stacking window manager. Specifically, is there a way to get workspace, window, or other relevant information through labwc's IPC? Any advice or workarounds would be helpful!

**Comment by stefonarch** [comment]
There is not and won't be any IPC,  afaik. 
For workspaces https://github.com/labwc/labwc/pull/2030 will be available at some point, so pagers can be implemented by panels. But you can build some tools/buttons using `wtype` for triggering shortcuts for actions.

Found the link: https://github.com/labwc/labwc-scope?tab=readme-ov-file#appendix-b---ipc

**Comment by winteriscariot** [comment]
while there's not an IPC, there are a number of sway/wlroots oriented tools that you can use that don't rely on sway-ipc. for instance, I use [wlrctl](https://github.com/DreamMaoMao/wlrctl) or [lswt](https://sr.ht/~leon_plickat/lswt) in combination with other tools to do various things around the labwc desktop. most things that are compositor independent and operate on wayland protocols should also work as needed.
